File: wearos/src/main/java/com/wishmarketplace/wearos/theme/WearTheme.kt
package com.wishmarketplace.wearos.theme

import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.wear.compose.material.MaterialTheme

/**
 * Wear OS color palette for Wish Marketplace
 * Optimized for AMOLED displays with true blacks for battery efficiency
 */
object WearColors {
    // Primary brand colors
    val Primary = Color(0xFF6366F1) // Indigo
    val PrimaryVariant = Color(0xFF4F46E5)
    val OnPrimary = Color(0xFFFFFFFF)

    // Secondary colors
    val Secondary = Color(0xFF10B981) // Green for success/winning bids
    val SecondaryVariant = Color(0xFF059669)
    val OnSecondary = Color(0xFFFFFFFF)

    // Background colors - true black for AMOLED
    val Background = Color(0xFF000000)
    val Surface = Color(0xFF1A1A1A)
    val SurfaceVariant = Color(0xFF2D2D2D)

    // Content colors
    val OnBackground = Color(0xFFE5E5E5)
    val OnSurface = Color(0xFFE5E5E5)
    val OnSurfaceVariant = Color(0xFFB3B3B3)

    // Status colors
    val Error = Color(0xFFEF4444)
    val Warning = Color(0xFFF59E0B)
    val Success = Color(0xFF10B981)
    val Info = Color(0xFF3B82F6)

    // Bidding specific colors
    val ActiveBid = Color(0xFF10B981)
    val OutbidRed = Color(0xFFEF4444)
    val WinningGold = Color(0xFFFBBF24)
}

/**
 * Typography scale optimized for Wear OS
 * All sizes account for small screen readability at arm's length
 */
object WearTypography {
    // Minimum font size should be 14sp for readability on small screens
    const val DisplayLarge = 24f
    const val DisplayMedium = 20f
    const val DisplaySmall = 18f

    const val TitleLarge = 18f
    const val TitleMedium = 16f
    const val TitleSmall = 14f

    const val BodyLarge = 16f
    const val BodyMedium = 14f
    const val BodySmall = 12f

    const val LabelLarge = 14f
    const val LabelMedium = 12f
    const val LabelSmall = 10f
}

/**
 * Spacing scale for Wear OS
 * Optimized for circular and square watch faces
 */
object WearSpacing {
    const val ExtraSmall = 4f
    const val Small = 8f
    const val Medium = 12f
    const val Large = 16f
    const val ExtraLarge = 24f
    const val ExtraExtraLarge = 32f

    // Minimum touch target size as per Wear OS guidelines
    const val MinTouchTarget = 48f
}

/**
 * Wear OS Material Theme wrapper
 * Applies consistent theming across all screens
 */
@Composable
fun WishWearTheme(
    content: @Composable () -> Unit
) {
    MaterialTheme(
        colors = androidx.wear.compose.material.Colors(
            primary = WearColors.Primary,
            primaryVariant = WearColors.PrimaryVariant,
            secondary = WearColors.Secondary,
            secondaryVariant = WearColors.SecondaryVariant,
            background = WearColors.Background,
            surface = WearColors.Surface,
            error = WearColors.Error,
            onPrimary = WearColors.OnPrimary,
            onSecondary = WearColors.OnSecondary,
            onBackground = WearColors.OnBackground,
            onSurface = WearColors.OnSurface,
            onSurfaceVariant = WearColors.OnSurfaceVariant,
            onError = Color.White
        ),
        content = content
    )
}

File: lib/features/wear/core/wear_theme.dart
import 'package:flutter/material.dart';

/// Wear OS color palette optimized for AMOLED displays
class WearColors {
  WearColors._();

  // Primary brand colors
  static const Color primary = Color(0xFF6366F1);
  static const Color primaryVariant = Color(0xFF4F46E5);
  static const Color onPrimary = Color(0xFFFFFFFF);

  // Secondary colors
  static const Color secondary = Color(0xFF10B981);
  static const Color secondaryVariant = Color(0xFF059669);
  static const Color onSecondary = Color(0xFFFFFFFF);

  // Background colors - true black for AMOLED
  static const Color background = Color(0xFF000000);
  static const Color surface = Color(0xFF1A1A1A);
  static const Color surfaceVariant = Color(0xFF2D2D2D);

  // Content colors
  static const Color onBackground = Color(0xFFE5E5E5);
  static const Color onSurface = Color(0xFFE5E5E5);
  static const Color onSurfaceVariant = Color(0xFFB3B3B3);

  // Status colors
  static const Color error = Color(0xFFEF4444);
  static const Color warning = Color(0xFFF59E0B);
  static const Color success = Color(0xFF10B981);
  static const Color info = Color(0xFF3B82F6);

  // Bidding specific colors
  static const Color activeBid = Color(0xFF10B981);
  static const Color outbidRed = Color(0xFFEF4444);
  static const Color winningGold = Color(0xFFFBBF24);
}

/// Theme data for Wear OS
class WearTheme {
  WearTheme._();

  static ThemeData get theme {
    return ThemeData(
      useMaterial3: true,
      brightness: Brightness.dark,
      scaffoldBackgroundColor: WearColors.background,
      colorScheme: const ColorScheme.dark(
        primary: WearColors.primary,
        onPrimary: WearColors.onPrimary,
        secondary: WearColors.secondary,
        onSecondary: WearColors.onSecondary,
        surface: WearColors.surface,
        onSurface: WearColors.onSurface,
        error: WearColors.error,
        onError: Colors.white,
      ),
      cardTheme: const CardThemeData(
        color: WearColors.surface,
        elevation: 0,
        margin: EdgeInsets.symmetric(vertical: 4),
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.all(Radius.circular(12)),
        ),
      ),
      textTheme: const TextTheme(
        displayLarge: TextStyle(fontSize: 24, fontWeight: FontWeight.bold, color: WearColors.onBackground),
        displayMedium: TextStyle(fontSize: 20, fontWeight: FontWeight.bold, color: WearColors.onBackground),
        displaySmall: TextStyle(fontSize: 18, fontWeight: FontWeight.bold, color: WearColors.onBackground),
        titleLarge: TextStyle(fontSize: 18, fontWeight: FontWeight.w600, color: WearColors.onSurface),
        titleMedium: TextStyle(fontSize: 16, fontWeight: FontWeight.w600, color: WearColors.onSurface),
        titleSmall: TextStyle(fontSize: 14, fontWeight: FontWeight.w600, color: WearColors.onSurface),
        bodyLarge: TextStyle(fontSize: 16, color: WearColors.onSurface),
        bodyMedium: TextStyle(fontSize: 14, color: WearColors.onSurface),
        bodySmall: TextStyle(fontSize: 12, color: WearColors.onSurfaceVariant),
        labelLarge: TextStyle(fontSize: 14, fontWeight: FontWeight.w500, color: WearColors.onSurface),
        labelMedium: TextStyle(fontSize: 12, fontWeight: FontWeight.w500, color: WearColors.onSurface),
        labelSmall: TextStyle(fontSize: 10, fontWeight: FontWeight.w500, color: WearColors.onSurfaceVariant),
      ),
      iconTheme: const IconThemeData(
        color: WearColors.onSurface,
        size: 24,
      ),
      elevatedButtonTheme: ElevatedButtonThemeData(
        style: ElevatedButton.styleFrom(
          minimumSize: const Size(48, 48),
          backgroundColor: WearColors.primary,
          foregroundColor: WearColors.onPrimary,
          shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(24)),
        ),
      ),
    );
  }
}
